src/runtime/network_monitor.hpp

```diff
@@ -0,0 +1,131 @@
+#pragma once
+
+#include <memory>
+#include <functional>
+#include <thread>
+#include <atomic>
+#include "platform_base.hpp"
+
+namespace runtime
+{
+  /**
+   * @enum NetworkStatus
+   * Represents the current network connectivity status.
+   */
+  enum class NetworkStatus
+  {
+    Online, // Network is available
+    Offline // Network is not available
+  };
+
+  /**
+   * @class NetworkMonitor
+   * Base class for monitoring network connectivity status across platforms.
+   * Provides a callback mechanism to notify when network status changes.
+   */
+  class NetworkMonitor
+  {
+  public:
+    using NetworkStatusCallback = std::function<void(NetworkStatus status)>;
+
+    NetworkMonitor() = default;
+    virtual ~NetworkMonitor() = default;
+
+    /**
+     * Start monitoring network status changes.
+     * @param callback Function to call when network status changes
+     * @return true if monitoring started successfully, false otherwise
+     */
+    virtual bool startMonitoring(NetworkStatusCallback callback) = 0;
+
+    /**
+     * Stop monitoring network status changes.
+     */
+    virtual void stopMonitoring() = 0;
+
+    /**
+     * Get the current network status.
+     * @return Current network connectivity status
+     */
+    virtual NetworkStatus getCurrentStatus() const = 0;
+
+    /**
+     * Create a platform-specific network monitor instance.
+     * @return Shared pointer to the appropriate NetworkMonitor implementation
+     */
+    static std::shared_ptr<NetworkMonitor> create();
+
+  protected:
+    NetworkStatusCallback statusCallback_;
+    bool isMonitoring_ = false;
+  };
+
+  /**
+   * @class DefaultNetworkMonitor
+   * Common implementation using socket-based connectivity detection.
+   * This can be used as a fallback or default implementation for platforms
+   * that don't have platform-specific network monitoring APIs.
+   */
+  class DefaultNetworkMonitor : public NetworkMonitor
+  {
+  public:
+    DefaultNetworkMonitor() = default;
+    ~DefaultNetworkMonitor() override;
+
+    bool startMonitoring(NetworkStatusCallback callback) override;
+    void stopMonitoring() override;
+    NetworkStatus getCurrentStatus() const override;
+
+  protected:
+    /**
+     * Test network connectivity using socket connection.
+     * @return true if network is available, false otherwise
+     */
+    bool testConnectivity() const;
+
+  private:
+    std::thread monitoringThread_;
+    std::atomic<bool> stopRequested_{false};
+  };
+
+#if UNITY_ANDROID
+  /**
+   * @class AndroidNetworkMonitor
+   * Android-specific implementation. For now, this uses the common socket-based approach
+   * since Android platform-specific APIs require Java/JNI integration which is not
+   * compatible with the C/C++ library constraints.
+   */
+  class AndroidNetworkMonitor : public DefaultNetworkMonitor
```

Comment:
Please use JNI to call `ConnectivityManager` or other related android APIs to implement the monitor instead of `DefaultNetworkMonitor`.